src/tribol/mesh/CouplingScheme.cpp

```diff
@@ -358,46 +360,55 @@ CouplingScheme::CouplingScheme( integer couplingSchemeId,
 
   m_loggingLevel = TRIBOL_UNDEFINED;
 
-  // STEP 0: create contact-pairs object associated with this coupling scheme
-  m_interfacePairs = new InterfacePairs( );
-
 } // end CouplingScheme::CouplingScheme()
 
 //------------------------------------------------------------------------------
-CouplingScheme::~CouplingScheme()
+void CouplingScheme::updateMeshViews()
 {
-  delete m_interfacePairs;
-  delete m_methodData;
+  auto mesh1 = MeshManager::getInstance().findData(m_mesh_id1);
+  auto mesh2 = MeshManager::getInstance().findData(m_mesh_id2);
+  SLIC_ERROR_ROOT_IF(mesh1 == nullptr || mesh2 == nullptr, 
+    "Register meshes before updating mesh views.");
+  m_mesh1 = mesh1->getView();
+  m_mesh2 = mesh2->getView();
 }
 
 //------------------------------------------------------------------------------
 bool CouplingScheme::isValidCouplingScheme()
 {
    bool valid {true};
    MeshManager & meshManager = MeshManager::getInstance(); 
-   if (!meshManager.hasMesh(this->m_meshId1) || !meshManager.hasMesh(this->m_meshId2))
+   if (!meshManager.findData(this->m_mesh_id1) || !meshManager.findData(this->m_mesh_id2))
    {
       SLIC_WARNING_ROOT("Please register meshes for coupling scheme, " << this->m_id << ".");
       return false;
    }
 
-   MeshData & mesh1 = meshManager.GetMeshInstance( this->m_meshId1 );
-   MeshData & mesh2 = meshManager.GetMeshInstance( this->m_meshId2 );
+   MeshData & mesh1 = meshManager.at( this->m_mesh_id1 );
+   MeshData & mesh2 = meshManager.at( this->m_mesh_id2 );
 
    // set boolean for null meshes
-   this->m_nullMeshes = mesh1.m_numCells <= 0 || mesh2.m_numCells <= 0;
+   this->m_nullMeshes = mesh1.numberOfElements() <= 0 || mesh2.numberOfElements() <= 0;
 
    // check for invalid mesh topology matches in a coupling scheme
-   if (mesh1.m_elementType != mesh2.m_elementType)
+   if (mesh1.getElementType() != mesh2.getElementType())
    {
-      SLIC_WARNING_ROOT("Coupling scheme, " << this->m_id << ", does not support meshes with " << 
+      SLIC_WARNING_ROOT("Coupling scheme " << this->m_id << " does not support meshes with " << 
                         "different surface element types.");
-      mesh1.m_isValid = false;
-      mesh2.m_isValid = false;
+      mesh1.isMeshValid() = false;
```

Comment:
We should refactor this so CouplingScheme doesn't change `MeshData::m_isValid`.  Otherwise, we can't define what `m_isValid` means in `MeshData` since anyone can go in and change it.  Perhaps define a bool on the coupling scheme which describes if the mesh is valid, independent of the MeshData value?